codeops-scripts/github-settings-definitions/default_branch_protection.ps1

```diff
@@ -0,0 +1,66 @@
+function default_branch_protection {
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory = $true)]
+        $SettingValue,
+
+        [Parameter(Mandatory = $true)]
+        [string] $Org,
+
+        [Parameter(Mandatory = $true)]
+        [string] $RepoName
+    )
+
+    if ($SettingValue -eq $true) {
+
+        # setup the payload for configuring branch protection
+        $body = @{
+            required_status_checks        = @{
+                strict   = $true
+                contexts = @()
+            }
+            enforce_admins                = $true
+            required_pull_request_reviews = @{
+                dismissal_restrictions     = @{
+                    users = @()
+                    teams = @()
+                }
+                dismiss_stale_reviews      = $true
+                require_code_owner_reviews = $false
+                # required_approving_review_count = 1           # requires preview API to change this
+            }
+            restrictions                  = @{
+                users = @()
+                teams = @()
+                apps  = @()
+            }
+        }
+
+        # We can only set branch protection on public repos
```

Comment:
How come? Is this a free vs paid restriction?

Reply:
Yes, that's right